sklearn/impute/_iterative.py

```diff
@@ -558,8 +558,12 @@ def fit_transform(self, X, y=None):
         else:
             self._estimator = clone(self.estimator)
 
-        if hasattr(self._estimator, 'random_state'):
-            self._estimator.random_state = self.random_state_
+        if (
+            hasattr(self._estimator, "random_state")
+            and self._estimator.random_state is None
+            and self.random_state is not None
+        ):
+            self._estimator.set_params(random_state=self.random_state)
```

Comment:
It was like that before, and I asked to change it. I don't think we should get a seed and pass around. If the estimator doesn't support rng, the user shouldn't pass an rng. I don't see why
we should workaround the API limitations of third party libraries.

Reply:
I am happy with the current approach, where the int should be used if `self._estimator` only supports random_state ints.

Reply:
Sorry for bringing this point late. I am wondering if `IterativeImputer` should be setting the `random_state` in `self._estimator` at all. In the example the user can do:

```py
imputer = IterativeImputer(LGBMRegressor(random_state=42), random_state = 456)
```

In this version, `IterativeImputer` would not set `self._estimator.random_state`.

What do you think @david-cortes ?

Reply:
I guess it's also a very valid approach (not setting the estimator's `random_state`), although it might cause some confusion when one has a seed/state set while the other doesn't. However in that snippet (setting the `random_state` in the estimator before passing it to the imputer), with the changes in this PR, it wouldn't set the estimator's `random_state` either, because it isn't `None`.

Reply:
When looking at metaestimators such as `RegressorChain` of `ClassifierChain`, it has a `random_state` parameter, but does not set the `random_state` of the `base_estimator`. I think not setting `random_state` would be more consistent with the rest of the library.